netbox/templates/extras/dashboard/widget.html

```diff
@@ -9,14 +9,14 @@
     gs-id="{{ widget.id }}"
 >
   <div class="card grid-stack-item-content">
-    <div class="card-header text-{{ widget.fg_color }} bg-{{ widget.color|default:"secondary" }} px-2 py-1 d-flex flex-row">
+    <div class="card-header {% if widget.color and widget.color != 'gray' %} text-{{ widget.fg_color }} {% endif %} bg-{{ widget.color|default:'default' }} px-2 py-1 d-flex flex-row">
```

Comment:
We want to avoid checking against specific values for maintainability reasons. It looks like the root issue might be related to #16385, which is being handled as a bug in the current release. If so, we can probably ignore this issue for now.

Reply:
The foreground color is [explicitly mapped](https://github.com/netbox-community/netbox/blob/22348cdbfc2b3b6ea200100704c03e3b82a3b63c/netbox/extras/dashboard/widgets.py#L135) to individual background colors; we can easily change these as needed.

Reply:
I've removed the 'if gray' statement in order for a better fix with #16385. I've kept in the if statement to see if a color is assigned or not in order for the header text color to change with the theme. Please let me know if there are any isues with this approach @jeremystretch